Implementations/Algorithms/Pong_REINFORCE.py

```diff
@@ -159,9 +217,49 @@ def set_seeds(self, s):
         torch.manual_seed(s)
         random.seed(s)
 
+class DiffFrame(gym.ObservationWrapper):
+    def __init__(self, env):
+        super().__init__(env)
+
+    @staticmethod
+    def prepro(I):
+        """ prepro 210x160x3 uint8 frame into 7840 (98x80) 1D float vector """
+        I = I[:195] # crop
+        I = I[::2,::2,0] # downsample by factor of 2
+        I[I == 144] = 0 # erase background (background type 1)
+        I[I == 109] = 0 # erase background (background type 2)
+        I[I != 0] = 1 # everything else (paddles, ball) just set to 1
+        return I
+
+    def observation(self, obs):
+        obs = np.array(obs)
+        I, J = DiffFrame.prepro(obs[:,:,0:3]), DiffFrame.prepro(obs[:,:,3:6])
+        return ((J-I).astype(np.float).ravel()+1)/2.0
+
+def wrap_diff(env, episode_life=True, clip_rewards=True, frame_stack=True, scale=True):
+    """Configure environment for DeepMind-style Atari.
+    """
+    # if episode_life:
+        # env = atari_wrappers.EpisodicLifeEnv(env)
+    # if scale:
+        # env = atari_wrappers.ScaledFloatFrame(env)
+    # if clip_rewards:
+        # env = atari_wrappers.ClipRewardEnv(env)
+    if frame_stack:
+        env = atari_wrappers.FrameStack(env, 2)
+    return DiffFrame(env)
+
+
+WIDTH = 98
+HEIGHT = 80
```

Comment:
For organzation sake it would be better to keep the constants together at the begining of the file.